src/app/dashboard/dashboard.tsx

```diff
@@ -14,7 +14,19 @@ export async function Dashboard({
   TESTING_FLAG?: boolean;
 }) {
   const user = (await dataModel.getUser(userID)) as User;
-  const userBudgets = await dataModel.getBudgetsForUser(userID);
+  const sgaUser = userIsSGA(user);
+  const budgets = await dataModel.getBudgets(
+    sgaUser ? [] : [new Filter('user_id', '==', userID)]
+  );
+
+  // Add organizer name to each budget
+  for (const budget of budgets) {
+    if (sgaUser) {
+      const organizer = await dataModel.getUser(budget.user_id);
+      budget.organizer = organizer.name;
+    } else budget.organizer = '';
+  }
```

Comment:
This loop makes many repetead calls to firebase and is probably a sign that we shoud just save the RSO's user name to the budget at the time of creation in the `create_budget_form`